sdk/react/widgets/SwitchButton.js

```diff
@@ -0,0 +1,72 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {Animated, TouchableHighlight} from 'react-native';
+
+import {STRING_CONSTANTS} from '../constants';
+
+const AccessibilityUtils = require('../accessibilityUtils');
+const styles = require('../utils').getStyles(require('./style/RectButtonStyles.json'));
+const Log = require('../log');
+
+class SwitchButton extends React.Component {
+
+  static propTypes = {
+    visible: PropTypes.bool.isRequired,
+    disabled: PropTypes.bool.isRequired,
+    isForward: PropTypes.bool.isRequired,
+    onSwitch: PropTypes.func.isRequired,
+    icon: PropTypes.string.isRequired,
+    fontStyle: PropTypes.object,
+    sizeStyle: PropTypes.object,
+    opacity: PropTypes.object,
+    animate: PropTypes.object,
+    buttonColor: PropTypes.object,
+  };
+
+  state = {
+    movedPosition: 0
+  };
+
+  componentWillReceiveProps(nextProps) {
+    this.state.movedPosition = nextProps.currentPosition;
```

Comment:
don't mutate the state directly, use setState
also `componentWillReceiveProps` is a wrong place to set state, please read the docs: https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops

Reply:
deleted because it isn't used